karpenter-operator/controllers/karpenter/karpenter_controller.go

```diff
@@ -126,13 +144,43 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, err
 	}
 	if hcp.DeletionTimestamp != nil {
-		// TODO(alberto): implement deletion. E.g. loop over nodeClaims delete them, wait and delete karpeneter deployment.
 		if controllerutil.ContainsFinalizer(hcp, karpenterFinalizer) {
+			nodePoolList := &karpenterv1.NodePoolList{}
+			if err := r.GuestClient.List(ctx, nodePoolList, client.InNamespace(r.Namespace)); err != nil {
```

Comment:
why are we using `client.InNamespace(r.Namespace)` here? this will list nodePools in the HCP namespace which doesn't exist in the guest cluster.

Reply:
My bad, thanks for catching that.